server-spi/src/main/java/org/keycloak/models/credential/dto/PasswordSecretData.java

```diff
@@ -15,26 +15,22 @@
     private final byte[] salt;
 
     @JsonCreator
-    public PasswordSecretData(@JsonProperty("value") String value, @JsonProperty("salt") String salt) {
-        this(value, decodeSalt(salt));
+    public PasswordSecretData(@JsonProperty("value") String value, @JsonProperty("salt") String salt) throws IOException {
+        if ("__SALT__".equals(salt)) {
```

Comment:
There is also the case when salt can be null. It can happen for custom hashing algorithms (and some people are using them). So could we change this line to "if (salt == null || "__SALT__".equals(salt))" ?

Reply:
Sure, it is valid case.

Reply:
I have updated my commit.